### PR TITLE
Implement RTL5b and RTL5j

### PR DESCRIPTION
**Pull request by owenpearson**
Spec alignment issue reported internally; see slack thread: https://ably-real-time.slack.com/archives/C01K3K2V5HB/p1627940634005500

Adds/modifies switch cases for detaching from channels in the `suspended` and `failed` states, along with associated tests.

### CODE REVIEW

File: spec/realtime/channel.test.js [hunk 1]
```diff
@@ -1506,5 +1506,35 @@ define(['ably', 'shared_helper', 'async', 'chai'], function (Ably, helper, async
...
 			);
 		});
+
+                // RTL5j
+                it('detaching from suspended channel transitions channel to detached state', function (done) {
+                    var realtime = helper.AblyRealtime({ transports: [helper.bestTransport] });
+                    var channelName = 'detach_from_suspended';
+                    var channel = realtime.channels.get(channelName);
+
+                    channel.state = 'suspended';
```
Comment (QuintinWillison) [inline spec/realtime/channel.test.js:1516]: Just to ensure my understanding: I'm assuming this isn't public API - i.e. the test is mutating an internal object?

**Review by QuintinWillison — APPROVED** [review]

File: spec/realtime/channel.test.js (hunk 1 above)
Comment (owenpearson) [reply spec/realtime/channel.test.js:1516]: It certainly is part of the public API and is documented in the typescript interface definition, but I think it should be clear to developers that the intention is for this to be readonly and I would expect no one would actually try mutating it unless they're trying to mess with the lib internals.

Comment (SimonWoolf) [reply spec/realtime/channel.test.js:1516]: typescript has a `readonly` attribute modifier now, we can add that to the typings on things like this to make the intention even clearer

**Review by SimonWoolf — APPROVED** [review]
lgtm with comment.

I'm now thinking it probably makes sense to modify `(RTL5k) If the channel receives an ATTACHED message while in the DETACHING state, it should send a new DETACH message and remain in the DETACHING state` to also apply to the `detached` state. Imagine: client lib is in `detached` after an explicit detach() while in `suspended` after a failed attach, and then the server finally catches up and sends an ATTACHED; we want to enforce the user's stated preference and send a DETACH rather than just acquiesce and end up in the `attached` state after the user called detach(), wdyt?

File: common/lib/client/realtimechannel.js [hunk 2]
```diff
@@ -308,10 +308,16 @@ var RealtimeChannel = (function() {
 			return;
 		}
 		switch(this.state) {
+                        case 'suspended':
+                                this.requestState('detached');
```
[the remark this answers is not in the document]
Comment (owenpearson) [reply common/lib/client/realtimechannel.js]: thanks, resolved in e6cfd1b

**Pull request merged**